tensorflow/python/training/learning_rate_decay_test.py

```diff
@@ -215,5 +217,83 @@ def testBeyondEndWithCycle(self):
       self.assertAllClose(decayed_lr.eval(), expected, 1e-6)
 
 
+class ExponentialDecayTest(test_util.TensorFlowTestCase):
+
+  def testDecay(self):
+    initial_lr = 0.1
+    k = 10
+    decay_rate = 0.96
+    step = state_ops.variable_op([], dtypes.int32)
+    assign = state_ops.assign(step, 0)
+    add = state_ops.assign_add(step, 1)
+    with self.test_session():
+      assign.op.run()
+      for i in range(k+1):
+        decayed_lr = learning_rate_decay.natural_exp_decay(initial_lr, step,
+                                                           k, decay_rate)
+        self.assertAllClose(initial_lr * math.exp(-i / k * decay_rate),
+                            decayed_lr.eval())
+        add.op.run()
+
+  def testStaircase(self):
+    initial_lr = 0.1
+    k = 10
+    decay_rate = 0.96
+    step = state_ops.variable_op([], dtypes.int32)
+    assign_step = state_ops.assign(step, 0)
+    increment_step = state_ops.assign_add(step, 1)
+    decayed_lr = learning_rate_decay.natural_exp_decay(initial_lr,
+                                                       step,
+                                                       k,
+                                                       decay_rate,
+                                                       staircase=True)
+    with self.test_session():
```

Comment:
@vrv please try this and let me know if there are any problems!  Thanks for your patience in dealing with this PR.